agent/eni/watcher_linux.go

```diff
@@ -131,8 +124,48 @@ func (udevWatcher *UdevWatcher) reconcileOnce() {
 	// As we postulate the netlinkClient.LinkList() call to be expensive, we allow
 	// the race here. The state would be corrected during the next reconciliation loop.
 
-	udevWatcher.state.Reconcile(currentState)
-	log.Debug("Udev watcher reconciliation: end")
+	// Add new interfaces next
+	for mac, _ := range currentState {
+		udevWatcher.sendENIStateChange(mac)
+	}
+	return nil
+}
+
+// sendENIStateChange handles the eni event from udev or reconcile phase
+func (udevWatcher *UdevWatcher) sendENIStateChange(mac string) {
+	eniAttachment, ok := udevWatcher.shouldSendENIStateChange(mac)
+	if ok {
+		go func(eni *api.ENIAttachment) {
+			eni.Status = api.ENIAttached
+			log.Infof("Emitting ENI change event for: %v", eni)
+			udevWatcher.eniChangeEvent <- api.TaskStateChange{
+				TaskArn:     eni.TaskArn,
+				Attachments: eni,
+			}
+		}(eniAttachment)
+	}
+}
+
+// shouldSendENIStateChange checks whether this eni is managed by ecs
+// and if its status should be sent to backend
+func (udevWatcher *UdevWatcher) shouldSendENIStateChange(macAddress string) (*api.ENIAttachment, bool) {
+	if macAddress == "" {
+		log.Warn("ENI state manager: device with empty mac address")
+		return nil, false
+	}
+	// check if this is an eni required by a task
+	eni, ok := udevWatcher.agentState.ENIByMac(macAddress)
+	if !ok {
+		log.Infof("ENI state manager: eni not managed by ecs: %s", macAddress)
```

Comment:
Yeah, I want to more here to reduce the verbosity of logs in this package. For example, do not scan non `ethernet` devices, ignore the primary ENI etc. Will follow up with another PR.